h2o-algos/src/main/java/hex/tree/SharedTreeModelWithContributions.java

```diff
@@ -190,11 +190,15 @@ protected void addContribToNewChunk(float[] contribs, NewChunk[] nc) {
 
   public class ScoreContributionsSortingTask extends ScoreContributionsTask {
 
-    private transient ContributionsOptions _options;
+    private int _topN;
+    private int _bottomN;
+    private boolean _abs;
 
-    public ScoreContributionsSortingTask(SharedTreeModel model, ContributionsOptions options) {
+    public ScoreContributionsSortingTask(SharedTreeModel model, int topN, int bottomN, boolean abs) {
```

Comment:
you can still accept ContributionsOptions in the constructor, just extract the fields you need here

this is a better design because of better encapsulation of logic in the class where it belongs

Reply:
True, thanks